src/main/java/org/opentripplanner/model/plan/Place.java

```diff
@@ -90,10 +92,20 @@ public static Place forStop(StopLocation stop) {
   }
 
   public static Place forFlexStop(StopLocation stop, Vertex vertex) {
+    var id = stop.getId().toString();
+    var name = stop.getName();
+    var hasUselessName = Optional
+      .ofNullable(name)
+      .map(I18NString::toString)
+      .map(s -> s.equals(id))
```

Comment:
What if the id and name are the same? I’ve seen some feeds, usually created by hand, where they are the same.

Reply:
This would still work because we would compare the name without the feed id with the id that has the feed id.

Only when the stop has no name then the name is set to the id with feed id:
https://github.com/opentripplanner/OpenTripPlanner/blob/118101d9177024497489f649dee48df6944a5237/src/main/java/org/opentripplanner/gtfs/mapping/LocationMapper.java#L38

However, I do agree that it's not the best way to model the information. What do you think about setting the name to null if there is no name?